src/DeviceInterface/InterfacePrototype.js

```diff
@@ -714,8 +713,10 @@ class InterfacePrototype {
                 domain: window.location.hostname
             })
 
+            const username = form.getValues(false).credentials?.username || ''
+
             // append the new credential to the topContextData so that the top autofill can display it
-            topContextData.credentials = [fromPassword(password)]
+            topContextData.credentials = [fromPassword(password, username)]
```

Comment:
We're now including the username so the native layer can make decisions accordingly (in this case it's for form saving).